drivers/sensor/tmp116/tmp116.c

```diff
@@ -46,6 +51,95 @@ static int tmp116_reg_write(const struct device *dev, uint8_t reg,
 			cfg->i2c_addr);
 }
 
+static bool check_eeprom_bound(const struct device *dev, off_t offset,
+			       size_t len)
+{
+	struct tmp116_data *drv_data = dev->data;
+
+	if ((offset + len) > EEPROM_TMP116_SIZE ||
+	    offset % EEPROM_SIZE_REG != 0 ||
+	    len % EEPROM_SIZE_REG != 0) {
+		return false;
+	}
+
+	/* TMP117 uses EEPROM[2] as temperature offset register */
+	if (drv_data->id == TMP117_DEVICE_ID &&
```

Comment:
I'm not sure about this part. Is it better to do as if the TMP117 have a contiguous EEPROM of 3*16bits instead of 4*16bits with reserved 16bits